### PR TITLE
Dilin refactor

### PR DESCRIPTION
**Pull request by dilin-MS**
1. Extract duplicate code.
2. Fix backward compatibility.

### CODE REVIEW

**Comment by ArthurMa1978** [comment]
Don't checkin until release the public preview version

File: src/AzureOperator.ts [hunk 1]
```diff
@@ -6,46 +6,18 @@
...
-    if (iotProject === undefined) {
-      await handleIoTWorkspaceProjectFolder(telemetryContext);
-      return;
-    }
-
-    const result = await iotProject.load();
-    if (!result) {
-      await askAndNewProject(telemetryContext);
+    const iotProject =
+        await createAndLoadIoTProject(context, channel, telemetryContext);
```
Comment (ArthurMa1978) [inline src/AzureOperator.ts]: What's the naming convention used in our extension? The first character is lower case or upper case? I see both in our code, should be unified.

Comment (ArthurMa1978) [reply src/AzureOperator.ts]: Does provision and deploy need create project?

File: src/AzureOperator.ts [hunk 2]
```diff
@@ -68,24 +40,9 @@ export class AzureOperator {
...
```
Comment (ArthurMa1978) [inline src/AzureOperator.ts]: Does deploy need create project?

File: src/AzureOperator.ts [hunk 3]
```diff
@@ -6,46 +6,18 @@
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
 import {TelemetryContext} from './telemetry';
-import {ProjectHostType} from './Models/Interfaces/ProjectHostType';
-import {handleIoTWorkspaceProjectFolder, askAndNewProject} from './utils';
+import {createAndLoadIoTProject} from './utils';
```
Comment (ArthurMa1978) [inline src/AzureOperator.ts]: Why not import all?

File: src/DeviceOperator.ts [hunk 4]
```diff
@@ -7,43 +7,15 @@
...
-import {ProjectHostType} from './Models/Interfaces/ProjectHostType';
-import {askAndNewProject, handleIoTWorkspaceProjectFolder} from './utils';
-
-const impor = require('impor')(__dirname);
-const ioTWorkspaceProjectModule = impor('./Models/IoTWorkspaceProject') as
-    typeof import('./Models/IoTWorkspaceProject');
-const ioTContainerizedProjectModule =
-    impor('./Models/IoTContainerizedProject') as
-    typeof import('./Models/IoTContainerizedProject');
+import {createAndLoadIoTProject} from './utils';
```
Comment (ArthurMa1978) [inline src/DeviceOperator.ts]: See my comments in AzureOperator.ts

File: src/AzureOperator.ts (hunk 1 above)
Comment (dilin-MS) [reply src/AzureOperator.ts]: Will fix.

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 5]
```diff
@@ -66,10 +66,8 @@ export class CodeGeneratorCore {
   async GenerateDeviceCodeStub(
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext): Promise<boolean> {
-    if (RemoteExtension.isRemote(context)) {
-      const message =
-          `The project is open in a Docker container now. Open a new window and run this command again.`;
-      vscode.window.showWarningMessage(message);
+    const notRemote = RemoteExtension.checkNotRemoteBeforeRunCommand(context);
+    if (!notRemote) {
```
Comment (ArthurMa1978) [inline src/DigitalTwin/CodeGeneratorCore.ts]: print message in the output window?

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 6]
```diff
@@ -66,10 +66,8 @@ export class CodeGeneratorCore {
   async GenerateDeviceCodeStub(
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext): Promise<boolean> {
-    if (RemoteExtension.isRemote(context)) {
-      const message =
-          `The project is open in a Docker container now. Open a new window and run this command again.`;
-      vscode.window.showWarningMessage(message);
+    const notRemote = RemoteExtension.checkNotRemoteBeforeRunCommand(context);
+    if (!notRemote) {
       return true;
```
Comment (ArthurMa1978) [inline src/DigitalTwin/CodeGeneratorCore.ts]: return false

Comment (ArthurMa1978) [reply src/DigitalTwin/CodeGeneratorCore.ts]: warning message?

File: src/exampleExplorer.ts [hunk 7]
```diff
@@ -186,12 +186,11 @@ export class ExampleExplorer {
   async selectBoard(
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext) {
-    if (RemoteExtension.isRemote(context)) {
-      const message =
-          `The project is open in a Docker container now. Open a new window and run this command again.`;
-      vscode.window.showWarningMessage(message);
+    const notRemote = RemoteExtension.checkNotRemoteBeforeRunCommand(context);
+    if (!notRemote) {
       return;
```
Comment (ArthurMa1978) [inline src/exampleExplorer.ts]: warning message?

File: src/projectInitializer.ts [hunk 8]
```diff
@@ -34,10 +34,8 @@ export class ProjectInitializer {
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext) {
     // Only create project when not in remote environment
-    if (RemoteExtension.isRemote(context)) {
-      const message =
-          `The project is open in a Docker container now. Open a new window and run this command again.`;
-      vscode.window.showWarningMessage(message);
+    const notRemote = RemoteExtension.checkNotRemoteBeforeRunCommand(context);
+    if (!notRemote) {
       return;
```
Comment (ArthurMa1978) [inline src/projectInitializer.ts]: warning message

File: src/utils.ts [hunk 9]
```diff
@@ -12,10 +12,19 @@ import {AzureFunctionsLanguage, DependentExtensions, FileNames, GlobalConstants,
...
 import {CodeGenProjectType, DeviceConnectionType} from './DigitalTwin/DigitalTwinCodeGen/Interfaces/CodeGenerator';
 import {FileUtility} from './FileUtility';
+import {ProjectHostType} from './Models/Interfaces/ProjectHostType';
 import {ProjectTemplate, TemplateFileInfo} from './Models/Interfaces/ProjectTemplate';
 import {RemoteExtension} from './Models/RemoteExtension';
 import {TelemetryContext} from './telemetry';
 
+const impor = require('impor')(__dirname);
+import {IoTWorkbenchProjectBase} from './Models/IoTWorkbenchProjectBase';
+const ioTWorkspaceProjectModule = impor('./Models/IoTWorkspaceProject') as
```
Comment (ArthurMa1978) [inline src/utils.ts]: what's the benefit of using dynamic load here?

File: src/AzureOperator.ts (hunk 2 above)
Comment (dilin-MS) [reply src/AzureOperator.ts]: Will change the function name to `constructAndLoadIoTProject`.

File: src/AzureOperator.ts (hunk 3 above)
Comment (dilin-MS) [reply src/AzureOperator.ts]: WIll change to `import {handleIoTWorkspaceProjectFolder, askAndNewProject, createAndLoadIoTProject} from './utils';`

File: src/DeviceOperator.ts (hunk 4 above)
Comment (dilin-MS) [reply src/DeviceOperator.ts]: No need to import all. Only use some function in `utils.ts`.

**Pull request merged**